cmd/slsa-provenance/cli/options/generate.go

```diff
@@ -20,35 +21,57 @@ type GenerateOptions struct {
 	ExtraMaterials []string
 }
 
+const (
+	// ContextLen defines the context content limit.
+	ContextLen = 1024 * 1024 // 1 MB
+)
+
+var (
+	// EnvGithubContext holds the environment variable name for Github context.
+	EnvGithubContext = "GITHUB_CONTEXT"
+	// EnvRunnerContext holds the environment variable name for Runner context.
+	EnvRunnerContext = "RUNNER_CONTEXT"
+)
+
 // GetGitHubContext The '${github}' context value, retrieved in a GitHub workflow.
 func (o *GenerateOptions) GetGitHubContext() (*github.Context, error) {
-	if o.GitHubContext == "" {
-		return nil, RequiredFlagError("github-context")
-	}
-	decodedContext, err := base64.StdEncoding.DecodeString(o.GitHubContext)
-	if err != nil {
-		return nil, err
+	// Retrieve context by environment
+	githubContext := os.Getenv(EnvGithubContext)
+	if githubContext == "" {
+		return nil, RequiredEnvironmentVariableError(EnvGithubContext)
 	}
+
+	// 1MB should be more than enough
+	lr := io.LimitReader(strings.NewReader(githubContext), ContextLen)
```

Comment:
> Which problem does it resolve?

Resource exhaustion via an uncontrolled input size. To decode the JSON, golang allocate a buffer, this buffer must have a limit.

> What if for some reason the context is bigger? That would break all those builds.

Yes, that's the purpose of this limiter. I chose 1MB arbitrary, potentially we could increase to 5MB but it looks unrealistic for me.